disnake/interactions/base.py

```diff
@@ -163,7 +163,10 @@ def _from_data(self, data: InteractionPayload):
             except KeyError:
                 pass
             else:
-                self.author = Member(state=self._state, guild=guild, data=member)  # type: ignore
+                author_id = data["member"]["user"]["id"]
+                self.author = guild.get_member(int(author_id)) or Member(
+                    state=self._state, guild=guild, data=member
+                )  # type: ignore
```

Comment:
```suggestion
                author_id = member["user"]["id"]
                self.author = guild.get_member(int(author_id)) or Member(
                    state=self._state, guild=guild, data=member  # type: ignore
                )
```

`guild` can be an `Object` here, need to account for that.